driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -19,46 +19,128 @@
+        return this.multiply(Expressions.numberToExpression(other));
     }
 
-    NumberExpression divide(NumberExpression n);
-
-    default NumberExpression divide(final Number divide) {
-        return this.divide(Expressions.numberToExpression(divide));
+    /**
+     * The result of dividing {@code this} value by the {@code other} value.
+     * This is not integer division: dividing {@code 1} by {@code 2} will yield
+     * {@code 0.5}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression divide(NumberExpression other);
+
+    /**
+     * The result of dividing {@code this} value by the {@code other} value.
+     * This is not integer division: dividing {@code 1} by {@code 2} will yield
+     * {@code 0.5}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default NumberExpression divide(final Number other) {
+        return this.divide(Expressions.numberToExpression(other));
     }
 
-    NumberExpression add(NumberExpression n);
-
-    default NumberExpression add(final Number add) {
-        return this.add(Expressions.numberToExpression(add));
+    /**
+     * The sum of adding {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression add(NumberExpression other);
+
+    /**
+     * The sum of adding {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default NumberExpression add(final Number other) {
+        return this.add(Expressions.numberToExpression(other));
     }
 
-    NumberExpression subtract(NumberExpression n);
-
-    default NumberExpression subtract(final Number subtract) {
-        return this.subtract(Expressions.numberToExpression(subtract));
+    /**
+     * The result of subtracting the {@code other} value from {@code this}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression subtract(NumberExpression other);
+
+    /**
+     * The result of subtracting the {@code other} value from {@code this}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default NumberExpression subtract(final Number other) {
+        return this.subtract(Expressions.numberToExpression(other));
     }
 
-    NumberExpression max(NumberExpression n);
-
-    NumberExpression min(NumberExpression n);
-
+    /**
+     * The larger value of {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression max(NumberExpression other);
+
+    /**
+     * The smaller value of {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression min(NumberExpression other);
```

Comment:
Should "larger"/"smaller" be hyperlinks to "gt"/"lt"? The same is applicable to `IntegerExpression`.

```java
{@linkplain #gt(Expression) larger}
{@linkplain #lt(Expression) smaller}
```